src/sentry/static/sentry/app/views/organizationIntegrations/integrationRepos.jsx

```diff
@@ -46,6 +47,36 @@ export default class IntegrationRepos extends AsyncComponent {
     return this.state.itemList.filter(repo => repo.integrationId === integrationId);
   }
 
+  debouncedSearchRepositoriesRequest = debounce(query => {
+    this.setState(
+      {
+        dropdownBusy: true,
+      },
+      () => this.searchRepositoriesRequest(query)
+    );
+  }, 200)
+
+  searchRepositoriesRequest = (searchQuery) => {
+    let orgId = this.context.organization.slug;
+    let query = {'search': searchQuery};
+    let endpoint = `/organizations/${orgId}/integrations/${this.props.integration.id}/repos/`;
+    return this.api.request(endpoint, {
+      method: 'GET',
+      query,
+      success: (data, _, jqXHR) => {
+        this.setState({integrationRepos: data, dropdownBusy: false});
+      },
+      error: error => {
+        this.setState({dropdownBusy: false});
+      },
+    });
+  }
+
+  handleSearchRepositories(e) {
+    this.setState({dropdownBusy: true});
+    this.debouncedSearchRepositoriesRequest(e.target.value);
+  }
```

Comment:
I think this can probably just turn into one method:

```js
handleSearchRepositories = debounce(e =>{
    this.setState({dropdownBusy: true});
    let orgId = this.context.organization.slug;
    let query = {'search': searchQuery};
    let endpoint = `/organizations/${orgId}/integrations/${this.props.integration.id}/repos/`;
    return this.api.request(endpoint, {
      method: 'GET',
      query,
      success: (data, _, jqXHR) => {
        this.setState({integrationRepos: data, dropdownBusy: false});
      },
      error: error => {
        this.setState({dropdownBusy: false});
      },
    });
}, 200)
```